main.go

```diff
@@ -14,191 +18,340 @@ import (
 	"github.com/prometheus/client_golang/prometheus"
 )
 
+const concurrentFetch = 100
+
+// Commandline flags.
 var (
-	configFile  = flag.String("config.file", "", "Path to config file.")
-	metricsPath = flag.String("web.telemetry-path", "/metrics", "Path under which to expose metrics.")
+	addr           = flag.String("web.listen-address", ":9105", "Address to listen on for web interface and telemetry")
+	autoDiscover   = flag.Bool("exporter.discovery", false, "Discover all Mesos slaves")
+	localAddr      = flag.String("exporter.local-address", "http://127.0.0.1:5051", "URL to the local Mesos slave")
```

Comment:
Basically, when connecting over HTTP(s) to something, I'd always use a full URL. Addresses are mainly used for listening on an ip:port yourself or for opening raw UDP/TCP connections.